### DISCUSSION TITLE
Problem with shift limits

### DISCUSSION THREAD

**Discussion by magentan0000** [reinterpretcat/vrp#109] [Q&A] [answered]
Hello together,

I'm currently working on a project for my university, and I can't figure out the time limit for shifts. Any help would be appreciated. 

I tried using "maxDuration" but it won't recognize the limit. Since I'm using an older version (before v1.12.0) I figured I would need to name it "shiftTime".  The shift limited is supposed to be 8 hours.

But "shiftTime" produces a lot of unassigned jobs, even tho I have enough vehicles. 

I'm a newbie to GitHub so I couldn't figure out how to upload a json file, so here is just an outtake. If you need the whole file, just let me know how to upload it. 

{
        "typeId": "truckbig",
        "vehicleIds": [
          "big_1",
          "big_2",
          "big_3"
        ],
        "profile": {
          "matrix": "truck"
        },
        "costs": {
          "fixed": 400.0,
          "distance": 0.0005,
          "time": 0.00
        },
        "shifts": [
          {
            "start": {
              "earliest": "2023-06-01T00:00:01Z",
              "location": {
                "lat": 48.85152,
                "lng": 9.1174
              }
            },
            "end": {
              "latest": "2023-06-01T23:59:59Z",
              "location": {
                "lat": 48.85152,
                "lng": 9.1174
              }
            }
          }
        ],
        "capacity": [
          25000
        ],
        "limits": {
          "shiftTime": 28800
        }
      }

Many thanks in advanced!

**Comment by reinterpretcat** [comment]
Hi, try to drag and drop files on the page as suggested below input field:
![image](https://github.com/reinterpretcat/vrp/assets/1611077/c3fd8d81-3251-455d-8820-46f9076ab8a0)

**Reply by magentan0000** [reply]
I uploaded it as a ZIP

**Reply by reinterpretcat** [reply, marked as answer]
Somehow I don't see the file..
But, I think I know what might be a problem: you rely on departure time optimization which shifts departure to a later time. However, it is lazy and requires some jobs to be assigned first. Maybe try to adjust start time of the vehicle (at the moment, it is at midnight)

**Reply by magentan0000** [reply]
Thank you very much, that helped a lot. A combination of later departure time and shift limit worked.

**Reply by MatthewHepburn** [reply]
I'm running into a similar problem, where jobs which could be completed if a vehicle left after its earliest departure time don't get routed if the vehicle has a `maxDuration` limit.

In my case I have jobs spread throughout the day, so moving the earliest departure time forward doesn't really help.
I tried creating duplicate vehicles to represent a range of possible start times, but this led to quite poor results.
